### DISCUSSION TITLE
Place numbering after content of heading.

### DISCUSSION THREAD

**Discussion by N4tus** [typst/typst#2479] [Q&A] [answered]
with `#set heading(numbering: "1.")` I can change how the numbering is shown, but I also want to change the order of the section content and the numbering, such that the numbering appears after the content.
I tried to use a show rule but it does not have the information of the heading numbers.

**Comment by PgBiel** [comment, marked as answer]
Try this:

```typ
#set heading(numbering: "1.")
#show heading: it => {
  if it.numbering == none { return it }
  let numbers = counter(heading).at(it.location())
  let display-number = numbering(it.numbering, ..numbers)
  let body = it.body
  [*#body #display-number*]
}

= Hello world!

== Hi!
```

![image](https://github.com/typst/typst/assets/9021226/379cbebe-3e9c-4ca6-914f-57c158c6be5c)

**Reply by EpicEricEE** [reply]
You (and @frozolotl) can actually make this a bit shorter by using the `numbering` parameter for `display`:
```typ
#let number = counter(heading).display(it.numbering)
```

**Reply by PgBiel** [reply]
@EpicEricEE I wanted to explicitly use the original heading's location, but since we are applying a show rule on it, then I believe your statement is correct. (However, using .display() wouldn't work when working with that heading in anywhere other than a show rule, since then you'd be in a different location than the heading itself.)

**Reply by N4tus** [reply]
Thank you for the fast answer. I did try experimenting with show rules, but was unable to get access to the number counter. I did not know/forget that you can use the counter function to increment the counter.

**Comment by frozolotl** [comment]
It appears @PgBiel was faster than me, but here's my a bit less general solution anyway (along with trashy titles):

```typ
#set heading(numbering: "1.1")
#show heading.where(level: 1): it => block[
  #if it.numbering == none { return it }
  #let num = counter(heading).display(it.numbering)
  Volume #num: #it.body
]

= Zebra World
== Escape
#lorem(10)

== Savanna
#lorem(10)

= The Great Zebra
= Return of the Zebra
```

![the rendered output](https://github.com/typst/typst/assets/44589151/44d71b40-3abe-42b5-86cc-7be3b3f962ea)

EDIT: Used improvement suggested by @EpicEricEE
